meta-client/src/main/java/org/apache/helix/metaclient/recipes/lock/LockInfoSerializer.java

```diff
@@ -0,0 +1,31 @@
+package org.apache.helix.metaclient.recipes.lock;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.metaclient.datamodel.DataRecord;
+import org.apache.helix.zookeeper.datamodel.ZNRecord;
+import org.apache.helix.zookeeper.datamodel.serializer.ZNRecordSerializer;
+
+public class LockInfoSerializer extends ZNRecordSerializer {
+  @Override
+  public Object deserialize(byte[] bytes) {
+    return new LockInfo(new DataRecord((ZNRecord) super.deserialize(bytes)));
```

Comment:
Could be exception here? Shall we throw it or catch it?

Reply:
Yes you're right, deserializing bytes if errors will return null which will throw an exception when creating the DataRecord. Will catch and log and return null. Thanks for the catch!